tests/integration/records/test_records_receivers.py

```diff
@@ -0,0 +1,269 @@
+                        'allow_push': allow_push,
+                    }
+                )
+            )
+
+
+def assert_db_has_no_user_record(user_record):
+    assert User.query.filter_by(email=user_record['email']).count() == 0
+    assert RemoteAccount.query.join(User).join(UserIdentity).filter(UserIdentity.id == user_record['orcid']).count() == 0
+    if 'token' in user_record:
+        assert RemoteToken.query.filter_by(access_token=user_record['token']).count() == 0
+
+    assert UserIdentity.query.filter_by(id=user_record['orcid']).count() == 0
+
+
+def cleanup_user_record(user_record):
+    if 'token' in user_record:
+        RemoteToken.query.filter_by(access_token=user_record['token']).delete()
+    user_id = db.session.query(UserIdentity.id_user).filter(UserIdentity.id == user_record['orcid']).subquery()
+    RemoteAccount.query.filter(RemoteAccount.user_id.in_(user_id)).delete(synchronize_session='fetch')
+    UserIdentity.query.filter_by(id=user_record['orcid']).delete()
+    User.query.filter_by(email=user_record['email']).delete()
+    assert_db_has_no_user_record(user_record)
+
+
+def assert_db_has_no_author_record(author_recid):
+    assert InspireRecord.query.filter_by().count() == 0
+
+
+def cleanup_author_record(author_recid):
+    author_record = get_db_record('aut', author_recid)
+    with db.session.nested():
+        db.session.delete(author_record)
+
+    assert_db_has_no_author_record(author_recid)
+
+
+@mock.patch('inspirehep.modules.records.receivers.Task')
+def test_orcid_push_not_trigger_for_author_records(mocked_Task, user_with_permission):
+    mocked_Task.assert_not_called()
+
+
+@mock.patch('inspirehep.modules.records.receivers.Task')
+def test_orcid_push_not_triggered_on_create_record_without_allow_push(mocked_Task, app, raw_record, user_without_permission):
+    migrate_and_insert_record(raw_record, skip_files=True)
+
+    mocked_Task.assert_not_called()
+
+
+@mock.patch('inspirehep.modules.records.receivers.Task')
+def test_orcid_push_not_triggered_on_create_record_without_token(mocked_Task, app, raw_record, user_without_token):
+    migrate_and_insert_record(raw_record, skip_files=True)
+
+    mocked_Task.assert_not_called()
+
+
+@mock.patch('inspirehep.modules.records.receivers.Task')
+def test_orcid_push_triggered_on_create_record_with_allow_push(mocked_Task, app, raw_record, user_with_permission):
+    mocked_Task.return_value = mocked_Task
+    migrate_and_insert_record(raw_record, skip_files=True)
+
+    expected_kwargs = {
+        'kwargs': {
+            'orcid': user_with_permission['orcid'],
+            'rec_id': 1608652,
+            'token': user_with_permission['token'],
+        },
+        'queue': 'orcid_push',
+    }
+
+    mocked_Task.apply_async.assert_called_once_with(**expected_kwargs)
+
+
+@mock.patch('inspirehep.modules.records.receivers.Task')
+def test_orcid_push_triggered_on_record_record_with_allow_push(mocked_Task, app, record, user_with_permission):
```

Comment:
Maybe renamerecord_record to record_update

Reply:
done